packages/client/src/lib.js

```diff
@@ -266,6 +238,63 @@ class NFTStorage {
     }
   }
 
+  /**
+   * @template {API.TokenInput} T
+   * @param {API.Encoded<T, [[Blob, Blob]]>} input
+   */
+  static async encodeNFT(input) {
+    validateERC1155(input)
+    return Token.Token.encode(input)
+  }
+
+  /**
+   * @param {Blob} blob
+   * @returns {Promise<{ cid: CID, car: import('./lib/interface.js').CarReader }>}
+   */
+  static async encodeBlob(blob) {
+    if (blob.size === 0) {
+      throw new Error('Content size is 0, make sure to provide some content')
+    }
+    const blockstore = new Blockstore()
+    const { root: cid } = await pack({
+      // @ts-ignore
+      input: [{ path: 'blob', content: blob.stream() }],
+      blockstore,
+      wrapWithDirectory: false,
+    })
+    const car = new BlockstoreCarReader(1, [cid], blockstore)
+    return { cid, car }
+  }
+
+  /**
+   * @param {Iterable<File>} files
+   * @returns {Promise<{ cid: CID, car: import('./lib/interface.js').CarReader }>}
+   */
+  static async encodeDirectory(files) {
+    const input = []
+    let size = 0
+    for (const file of files) {
+      input.push({ path: file.name, content: file.stream() })
+      size += file.size
+    }
+
+    if (size === 0) {
+      throw new Error(
+        'Total size of files should exceed 0, make sure to provide some content'
+      )
+    }
+
+    const blockstore = new Blockstore()
```

Comment:
NIt: It's second time I see following 7 lines of code only diff is `wrapWithDirectory` flag. I think it would be a good idea to factor it out to a function that just gives back `cid` and `car`.